### PR TITLE
added closing_angle; minor geometry modifications

### PR DESCRIPTION
**Pull request by smichr**
Is there a better name for `closing_angle` which represents the angle of one linear entity relative to another? Maybe `angle_from`? This is like the clockhand angle. `angle_between` always gives the smallest angle between to lines, but sometimes one wants to know the 4 quadrant angle.

Also: edits to docstrings and reordering of `Point.__new__` to put Point handling sooner.

### CODE REVIEW

File: sympy/geometry/line.py [hunk 1]
```diff
@@ -156,6 +157,50 @@ def angle_between(l1, l2):
...
+        if not isinstance(l1, LinearEntity) and not isinstance(l2, LinearEntity):
+            raise TypeError('Must pass only LinearEntity objects')
+
+        v1, v2 = l1.direction, l2.direction
+        ang = l1.angle_between(l2)
+        if not ang:
+            return ang
+        ok = not ang or v2.unit.rotate(ang).equals(v1.unit)
+        if ok is None:
+            raise Undecidable("can't determine orientation of %s and %s" % (l1, l2))
```
Comment (szymag) [inline sympy/geometry/line.py]: For what kind of situation `Undecidable` will be raised? You didn't add test for this line.

**Comment by szymag** [comment]
In my opinion `closing_angle` is good name  but I also recommend more detailed description. I was able to understand the code mainly from description in this PR not the dosctring.

**Comment by smichr** [comment]
I am going to make this a Ray property. Anyone wanting to treat something like a Ray can easily convert it to a Ray.

**Comment by smichr** [comment]
This should probably just be `atan2(*list(reversed(l2.direction - l1.direction)))`

**Comment by smichr** [comment]
ping @szymag

**Comment by szymag** [comment]
In my opinion it's ready to by merged.

**Pull request merged**